File: CinemaReservations/Java/CinemaAllocationsAcceptanceTests/src/test/java/com/sdd/cinemaallocationsacceptancetests/TicketBoothShould.java
package com.sdd.cinemaallocationsacceptancetests;

import com.sdd.cinemaallocations.*;
import com.sdd.cinemaallocationsacceptancetests.helpers.StubMovieScreeningRepository;
import org.junit.Test;

import java.io.IOException;

import static com.google.common.truth.Truth.assertThat;

public class TicketBoothShould {

    private static final String FORD_THEATER = "1";
    private static final String DOCK_STREET = "3";
    private static final String MADISON_THEATER = "5";

    MovieScreeningRepository repository = new StubMovieScreeningRepository();

    public TicketBoothShould() throws IOException {
    }

    @Test
    public void reserve_one_seat_when_available() {
        int partyRequested = 1;

        TicketBooth ticketBooth = new TicketBooth(repository);
        SeatsAllocated seatsAllocated = ticketBooth.allocateSeats(new AllocateSeats(FORD_THEATER, partyRequested));

        assertThat(seatsAllocated.reservedSeats()).hasSize(1);
        assertThat(seatsAllocated.reservedSeats().get(0).toString()).isEqualTo("A3");
    }

    @Test
    public void Reserve_multiple_seats_when_available() {
        int partyRequested = 3;

        TicketBooth ticketBooth = new TicketBooth(repository);
        SeatsAllocated seatsAllocated = ticketBooth.allocateSeats(new AllocateSeats(DOCK_STREET, partyRequested));

        assertThat(seatsAllocated.reservedSeats()).hasSize(3);
        assertThat(seatsAllocated.reservedSeats().get(0).toString()).isEqualTo("A6");
        assertThat(seatsAllocated.reservedSeats().get(1).toString()).isEqualTo("A7");
        assertThat(seatsAllocated.reservedSeats().get(2).toString()).isEqualTo("A8");
    }

    @Test
    public void return_SeatsNotAvailable_when_all_seats_are_unavailable() {
        int partyRequested = 1;

        TicketBooth ticketBooth = new TicketBooth(repository);
        SeatsAllocated seatsAllocated = ticketBooth.allocateSeats(new AllocateSeats(MADISON_THEATER, partyRequested));

        assertThat(seatsAllocated).isInstanceOf(NoPossibleAllocationsFound.class);
    }

    @Test
    public void return_TooManyTicketsRequested_when_9_tickets_are_requested() {
        int partyRequested = 9;

        TicketBooth ticketBooth = new TicketBooth(repository);
        SeatsAllocated seatsAllocated = ticketBooth.allocateSeats(new AllocateSeats(MADISON_THEATER, partyRequested));

        assertThat(seatsAllocated).isInstanceOf(TooManyTicketsRequested.class);
    }

}

File: CinemaReservations/CSharp/CinemaAllocations.Tests/TicketBoothShould.cs
using CinemaAllocations.Domain;
using CinemaAllocations.Tests.StubMovieScreening;
using NFluent;
using Xunit;

namespace CinemaAllocations.Tests
{
    public class SeatsAllocatorShould
    {
        private const string FordTheaterId = "1";
        private const string DockStreetId = "3";
        private const string MadisonTheatherId = "5";
        
        [Fact]
        public void Reserve_one_seat_when_available()
        {
            const int partyRequested = 1;

            IMovieScreeningRepository repository = new StubMovieScreeningRepository();
            var ticketBooth = new TicketBooth(repository);

            var seatsAllocated = ticketBooth.AllocateSeats(new AllocateSeats(FordTheaterId, partyRequested));

            Check.That(seatsAllocated.ReservedSeats).HasSize(1);
            Check.That(seatsAllocated.ReservedSeats[0].ToString()).IsEqualTo("A3");
        }
        
        [Fact]
        public void Reserve_multiple_seats_when_available()
        {
            const int partyRequested = 3;

            IMovieScreeningRepository repository = new StubMovieScreeningRepository();
            var ticketBooth = new TicketBooth(repository);

            var seatsAllocated = ticketBooth.AllocateSeats(new AllocateSeats(DockStreetId, partyRequested));

            Check.That(seatsAllocated.ReservedSeats).HasSize(3);
            Check.That(seatsAllocated.ReservedSeats[0].ToString()).IsEqualTo("A6");
            Check.That(seatsAllocated.ReservedSeats[1].ToString()).IsEqualTo("A7");
            Check.That(seatsAllocated.ReservedSeats[2].ToString()).IsEqualTo("A8");
        }

        [Fact]
        public void Return_SeatsNotAvailable_when_all_seats_are_unavailable()
        {
            const int partyRequested = 1;

            IMovieScreeningRepository repository = new StubMovieScreeningRepository();
            var ticketBooth = new TicketBooth(repository);

            var seatsAllocated = ticketBooth.AllocateSeats(new AllocateSeats(MadisonTheatherId, partyRequested));

            Check.That(seatsAllocated).IsInstanceOf<NoPossibleAllocationsFound>();
        }

        [Fact]
        public void Return_TooManyTicketsRequested_when_9_tickets_are_requested()
        {
            const int partyRequested = 9;

            IMovieScreeningRepository repository = new StubMovieScreeningRepository();
            var ticketBooth = new TicketBooth(repository);

            var seatsAllocated = ticketBooth.AllocateSeats(new AllocateSeats(MadisonTheatherId, partyRequested));
            
            Check.That(seatsAllocated).IsInstanceOf<TooManyTicketsRequested>();

        }
    }
}